ClosedXML/Excel/CalcEngine/Functions/DateAndTime.cs

```diff
@@ -113,6 +115,31 @@ private static object Date(List<Expression> p)
             return (int)Math.Floor(new DateTime(year, month, day).AddDays(daysAdjustment).ToOADate());
         }
 
+        private static object Datedif(List<Expression> p)
+        {
+            DateTime startDate = p[0];
+            DateTime endDate = p[1];
+            string unit = p[2];
+
+            if (startDate > endDate)
+                throw new NumberException("The start date is greater than the end date");
+
+            return (unit.ToUpper()) switch
+            {
+                "Y" => endDate.Year - startDate.Year - (new DateTime(startDate.Year, endDate.Month, endDate.Day) < startDate ? 1 : 0),
+                "M" => Math.Truncate((endDate.Year - startDate.Year) * 12d + endDate.Month - startDate.Month - (endDate.Day < startDate.Day ? 1 : 0)),
+                "D" => Math.Truncate(endDate.Subtract(startDate).TotalDays),
```

Comment:
Nope. 
![image](https://user-images.githubusercontent.com/19576939/99943763-410ce400-2d8b-11eb-999b-d584caf56510.png)

This should do the trick:
```suggestion
                "D" => Math.Truncate(endDate.Date.Subtract(startDate.Date).TotalDays),
```

Or even without `Math.Truncate` - CLR does not seem to care about daylight saving time and returns whole numbers anyway.